File: January/Day8/leetcode_149.cpp
class Solution {
public:
    int maxPoints(vector<vector<int>>& points) {
        int max_num = 10001;
        int count = 0;
        for(int i = 0; i<points.size()-1; i++)
        {
             unordered_map<double,int>mp;
            for(int j =i+1;j<points.size();j++)
            {
                if(points[i][0] == points[j][0])
                {
                   mp[10001]++;
                }
                else
                {
                    double slope = (double)(points[j][1]-points[i][1])/(double)(points[j][0]-points[i][0]);
                    mp[slope]++;
                }
            }
        for(auto x : mp)
        {
            count = max(count,x.second+1);
        }
        }
        return max(1,count);
    }
};

File: January/Day8/leetcode_149.java
class Solution {
    public int maxPoints(int[][] points) {
        int max=10001;
        int count=0;
        for(int i=0;i<points.length;i++){
            HashMap<Double,Integer> map= new HashMap<>();
            for(int j=i+1;j<points.length;j++){
                if(points[i][0]==points[j][0])map.put(10001.0,map.getOrDefault(10001.0,0)+1);
                else {
                    double slope= (points[j][1]-points[i][1])/(double)(points[j][0]-points[i][0]);
                    if(slope==-0.0)slope=0.0;

                    map.put(slope,map.getOrDefault(slope,0)+1);
                    System.out.println(slope);
                }
            }
            for(Map.Entry<Double,Integer> mapElement : map.entrySet()) {
            count=Math.max(count,mapElement.getValue()+1);
            }

        }
        return count==0?1:count;
        
    }
}
